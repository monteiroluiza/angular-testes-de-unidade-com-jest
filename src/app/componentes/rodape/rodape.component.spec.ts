import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RodapeComponent } from './rodape.component';

describe('RodapeComponent', () => {
  let component: RodapeComponent;
  let fixture: ComponentFixture<RodapeComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RodapeComponent],
    });
    fixture = TestBed.createComponent(RodapeComponent);
    component = fixture.componentInstance;
  });

  it('deveria definir as propriedades alt e src', () => {
    expect(component.src).toBeDefined();
    expect(component.alt).toBeDefined();
  });

  it('deveria renderizar o conteudo baseado nas propriedades src e alt', () => {
    component.alt = 'Imagem teste';
    component.src = 'https://example.com/test-image.jpg';
    expect(component).toMatchSnapshot();
  })
});
