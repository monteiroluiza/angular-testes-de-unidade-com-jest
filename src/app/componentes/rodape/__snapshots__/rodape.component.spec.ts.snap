// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RodapeComponent deveria renderizar o conteudo baseado nas propriedades src e alt 1`] = `
RodapeComponent {
  "__ngContext__": 2,
  "alt": "Imagem teste",
  "src": "https://example.com/test-image.jpg",
}
`;
